core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1030,8 +1034,22 @@ public String allocationId() {
         }
     }
 
+    /**
+     * The {@code ReplicasProxy} is an implementation of the {@code Replicas}
+     * interface that performs the actual {@code ReplicaRequest} on the replica
+     * shards. It also encapsulates the logic required for failing the replica
+     * if deemed necessary as well as marking it as stale when needed.
+     */
     final class ReplicasProxy implements ReplicationOperation.Replicas<ReplicaRequest> {
```

Comment:
I originally had them sharing knowledge (subclassing), however, `ReplicasProxy` used parts of `TransportReplicationAction` and made `WriteActionReplicasProxy` not able to subclass it well.